docs/report_identification.md

```diff
@@ -0,0 +1,24 @@
+# Analyzer report identification
+
+Unique report identifiers are required to compare analysis results.
+If the analyzers (`Clang Static Analyzer`, `Clang Tidy`) do not generate
+a unique report identifier, CodeChecker tries to generate one.
+
+It is recommended to use the identifier generated by the analyzers
+because they can include the semantic context of the result better into the
+unique id.
+
+## Experimental context-free hash generation method
+
+This method is used for Clang Tidy reports.
+The current hash generation algorithm in CodeChecker uses this information to
+generate the unique id.
+__NOTE: As the main diagnostic section the last element from the bug path is used!__
+
+Unique id content:
+
+  * `file_name` from the main diag section.
+  * `checker message`
+  * `line content` from the source file if can be read up
+  * `column numbers` from the main diag section
+  * whitespaces from the beginning of the source content are removed
```

Comment:
Since column number contain the leading white spaces, this is not so useful.

My suggestion:
Before calculating hash, replace every white space sequence by a singe space and recalculate the column number.
This way the hash will be resilient to indentation

The ultimate best would be to remove the trailing //... comments before hashing.

Reply:
duplicate whitespaces are removed and column numbers updated according to the changes. Trailing comments are not removed because detecting in python is hard without proper parsing support for C/C++.
In this case it is possible that the comment mark is detected as an end line comment even if it is not at the end of the line:
```
std::cout<<("some text //comment")
```